turbinia/workers/analysis/wordpress_creds.py

```diff
@@ -0,0 +1,192 @@
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+
+    # Where to store the resulting output file.
+    output_file_name = 'wordpress_creds_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    try:
+      location, num_files = self._collect_wordpress_file(evidence)
+      if num_files == 0:
+        result.close(self, success=True, status='No Wordpress database found')
+        return result
+    except TurbiniaException as e:
+      result.close(
+          self, success=True,
+          status='Error retrieving Wordpress database: {0:s}'.format(str(e)))
+      return result
+
+    try:
+      (creds, hashnames) = self._extract_wordpress_hashes(location)
+    except TurbiniaException as e:
+      result.close(self, success=True, status=str(e))
+      return result
+
+    (report, priority, summary) = self._analyse_wordpress_creds(
+        creds, hashnames)
+    output_evidence.text_data = report
+    result.report_data = report
+    result.report_priority = priority
+
+    # Write the report to the output file.
+    with open(output_file_path, 'wb') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+
+    return result
+
+  def _collect_wordpress_file(self, evidence):
+    """Extract artifacts using image_export.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+    Returns:
+        location (str): The file path to the extracted evidence.
+        number of artifacts (int): The number of files extracted.
+    """
+    try:
+      collected_artifacts = extract_files(
+          file_name=_WP_DB_NAME, disk_path=evidence.local_path,
+          output_dir=os.path.join(self.output_dir, 'artifacts'))
+    except TurbiniaException as e:
+      raise TurbiniaException('artifact extraction failed: {}'.format(str(e)))
+
+    # Extract base dir from our list of collected artifacts
+    location = os.path.dirname(collected_artifacts[0])
+
+    return (location, len(collected_artifacts))
+
+  def _extract_wordpress_hashes(self, location):
+    """Dump the Wordpress credentials from a raw database file.
+
+    Args:
+        location (list): Directory of files extracted from the disk.
+
+    Returns:
+        creds (list): List of strings containing raw extracted credentials
+        hashnames (dict): Dict mapping hash back to username for convenience.
+
+    Raises:
+      TurbiniaException: when the process fails.
+    """
+    creds = []
+    hashnames = {}
+    for dirs, _, files in os.walk(location):
+      if _WP_DB_NAME in files:
+        try:
+          strings = subprocess.Popen(
+              ['strings', os.path.join(dirs, _WP_DB_NAME)],
```

Comment:
Can we add this to the binary dependencies list in the config file?

Reply:
(might as well add `grep` too then)

Reply:
Done